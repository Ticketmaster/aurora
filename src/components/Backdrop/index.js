import React, { Component } from "react";
import PropTypes from "prop-types";

import { ESCAPE } from "../../utils/keyCharCodes";

import { Overlay } from "./index.styles";

export default class Backdrop extends Component {
  static propTypes = {
    /* eslint-disable-next-line react/forbid-prop-types */
    childRef: PropTypes.object.isRequired,
    children: PropTypes.node.isRequired,
    overlay: PropTypes.bool,
    isVisible: PropTypes.bool,
    animated: PropTypes.bool,
    /* eslint-disable-next-line react/forbid-prop-types */
    overlayProps: PropTypes.object,
    onRequestClose: PropTypes.func
  };

  static defaultProps = {
    overlay: true,
    overlayProps: {},
    onRequestClose: null,
    isVisible: true,
    animated: false
  };

  componentDidMount() {
    document.addEventListener("click", this.handleOutsideClick);
    document.addEventListener("keydown", this.onKeyPress);
  }

  componentWillUnmount() {
    document.removeEventListener("click", this.handleOutsideClick);
    document.removeEventListener("keydown", this.onKeyPress);
  }

  onKeyPress = e => {
    const { onRequestClose } = this.props;

    if (e.keyCode === ESCAPE && onRequestClose) {
      onRequestClose(e);
    }
  };

  handleOutsideClick = e => {
    const { childRef, onRequestClose } = this.props;

    if (
      onRequestClose &&
      childRef.current &&
      !childRef.current.contains(e.target)
    ) {
      onRequestClose(e);
    }
  };

  render() {
    const { children, overlay, overlayProps, isVisible, animated } = this.props;

    if (overlay) {
      return (
        <Overlay isVisible={isVisible} animated={animated} {...overlayProps}>
          {children}
        </Overlay>
      );
    }

    return children;
  }
}
