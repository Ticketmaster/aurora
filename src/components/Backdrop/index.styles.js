/* eslint-disable import/prefer-default-export */

import styled from "styled-components";
import zIndex from "../../theme/zIndex";

export const Overlay = styled.div.attrs({
  className: "container--overlay"
})`
  height: 100%;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  background-color: rgba(38, 38, 38, 0.5);
  z-index: ${zIndex.layout.overlay};
`;
