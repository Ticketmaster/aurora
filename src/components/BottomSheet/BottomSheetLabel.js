import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import colors from "../../theme/colors";

import { Text } from "../Text";
import { StatusBadge } from "../StatusBadge";

const LabelText = styled(Text)`
  text-transform: uppercase;
`;

const BottomSheetLabel = ({ variant, children, ...restProps }) => {
  if (!children) {
    return null;
  }

  switch (variant) {
    case "alert":
      return (
        <StatusBadge
          {...restProps}
          label={children}
          color={colors.alert.base}
        />
      );
    case "positive":
      return (
        <StatusBadge
          {...restProps}
          label={children}
          color={colors.positive.base}
        />
      );
    default:
      return (
        <LabelText {...restProps} primary>
          {children}
        </LabelText>
      );
  }
};

BottomSheetLabel.defaultProps = {
  variant: "default",
  children: ""
};

BottomSheetLabel.propTypes = {
  variant: PropTypes.oneOf(["default", "positive", "alert"]),
  children: PropTypes.string
};

export default BottomSheetLabel;
