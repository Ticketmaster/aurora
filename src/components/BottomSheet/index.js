import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import colors from "../../theme/colors";

import { Row } from "../Grid";
import CrossIcon from "../Icons/Cross";

const Backdrop = styled.div`
  height: 100%;
  width: 100%;
  position: fixed;
  z-index: 1300;
  left: 0;
  right: 0;
  top: 0;
  display: ${({ visible }) => (visible ? "flex" : "none")};
  background-color: rgba(0, 0, 0, 0.2);
  transition: opacity 5s ease-in-out, display 0.5s ease-in-out;
  opacity: ${({ visible }) => (visible ? "1" : "0")};
`;

const BottomSheetContent = styled.div`
  width: 100%;
  max-height: ${({ visible }) => (visible ? "calc(100% - 60px)" : "0")};
  position: fixed;
  bottom: 0;
  top: 60px;
  background-color: ${colors.white.base};
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  transition: max-height 0.5s cubic-bezier(0.4, 0, 0.2, 1);
  overflow-y: scroll;
`;

const ClickableButton = styled.button.attrs({
  className: "cancel"
})`
  background: none;
  border: none;
  cursor: pointer;
  justify-content: flex-end;
  align-items: flex-end;
  text-align: end;
  padding: 18px 18px 0 18px;
`;

const CancelButtonRow = styled(Row)`
  justify-content: flex-end;
  align-items: flex-end;
`;

class BottomSheet extends Component {
  componentDidMount() {
    document.addEventListener("click", this.handleOutsideClick, false);
  }

  componentWillUnmount() {
    document.removeEventListener("click", this.handleOutsideClick, false);
  }

  handleOutsideClick = e => {
    const { onCloseRequest, visible } = this.props;

    if (this.modal) {
      if (!this.modal.contains(e.target) && visible) {
        onCloseRequest();
        document.removeEventListener("click", this.handleOutsideClick, false);
      }
    }
  };

  render() {
    const { children, visible, onCloseRequest } = this.props;
    return (
      <Backdrop onClick={this.handleOutsideClick} visible={visible}>
        {visible ? (
          <BottomSheetContent
            innerRef={node => {
              this.modal = node;
            }}
            visible={visible}
            role="dialog"
            aria-modal
          >
            <CancelButtonRow>
              <ClickableButton
                aria-label="Close BottomSheet"
                role="button"
                onClick={onCloseRequest}
              >
                <CrossIcon size={12} color={colors.onyx.base} />
              </ClickableButton>
            </CancelButtonRow>
            {children}
          </BottomSheetContent>
        ) : null}
      </Backdrop>
    );
  }
}

BottomSheet.propTypes = {
  children: PropTypes.node.isRequired,
  visible: PropTypes.bool.isRequired,
  onCloseRequest: PropTypes.func.isRequired
};

export default BottomSheet;
