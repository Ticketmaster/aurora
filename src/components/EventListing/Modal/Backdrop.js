import styled from 'styled-components';
import Flex from "../../Flex";

const Backdrop = styled(Flex)`
  align-items: center;
  background-color: rgba(0,0,0,.5);
  height: 100%;
  justify-content: center;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 99; 
`;

export default Backdrop;
