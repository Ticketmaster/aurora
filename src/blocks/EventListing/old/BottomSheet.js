import React from "react";
import { TransitionGroup, CSSTransition } from "react-transition-group";

import Backdrop from "../../components/Backdrop";
import AuroraSheet from "../../components/BottomSheet";

const BottomSheet = (props) => (
<TransitionGroup component={Backdrop} appear>
  <CSSTransition
    classNames="bottom-sheet"
    timeout={300}
  >
    <AuroraSheet
      label={"On sale: MON \u2022 AUG 27 \u2022 10 AM"}
      labelVariant="positive"
      withCancelBtn={false}
    >
      {props.children}
    </AuroraSheet>
  </CSSTransition>
</TransitionGroup>
);

export default BottomSheet;
