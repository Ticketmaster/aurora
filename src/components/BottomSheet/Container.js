import React, { Component } from "react";
import PropTypes from "prop-types";

import BottomSheet from "./";

class BottomSheetContainer extends Component {
  state = { visible: true };

  onCloseRequest = () => {
    this.props.onClose();
    this.setState({ visible: !this.state.visible });
  };

  render() {
    const { children } = this.props;
    return (
      <BottomSheet
        visible={this.state.visible}
        onCloseRequest={this.onCloseRequest}
      >
        {children}
      </BottomSheet>
    );
  }
}

BottomSheetContainer.propTypes = {
  children: PropTypes.node.isRequired,
  onClose: PropTypes.func.isRequired
};

export default BottomSheetContainer;
