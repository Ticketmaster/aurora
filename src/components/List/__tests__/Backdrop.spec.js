import React from "react";
import { render, fireEvent } from "react-testing-library";

import Backdrop from "../Backdrop";
import BottomSheet from "../../BottomSheet";

jest.mock("../../List/Context", () => ({
  ItemContainerConsumer: jest
    .fn()
    .mockImplementation(({ children }) =>
      children({ onCloseRequest: jest.fn() })
    )
}));

describe("<Backdrop />", () => {
  it("renders Backdrop correctly", () => {
    const { container } = render(
      <Backdrop>
        <BottomSheet>
          <div>Europe</div>
          <div>Africa</div>
          <div>Asias</div>
        </BottomSheet>
      </Backdrop>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls the onKeyPress function on pressing the ESC key", () => {
    const { container } = render(
      <Backdrop>
        <BottomSheet>
          <div>Europe</div>
          <div>Africa</div>
          <div>Asias</div>
        </BottomSheet>
      </Backdrop>
    );

    fireEvent.keyDown(global.document, {
      key: "Escape",
      keyCode: 27
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls the handleOutsideClick function on clicking overlay", () => {
    const { container } = render(
      <Backdrop data-testid="container--overlay">
        <div>Europe</div>
      </Backdrop>
    );

    fireEvent.click(global.document);
    expect(container.firstChild).toMatchSnapshot();
  });
});
