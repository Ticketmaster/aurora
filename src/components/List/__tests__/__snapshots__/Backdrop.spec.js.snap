// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> calls the handleOutsideClick function on clicking overlay 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  background-color: rgba(0,0,0,0.75);
  z-index: 4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  class="c0 container--overlay"
  data-testid="container--overlay"
>
  <div>
    Europe
  </div>
</div>
`;

exports[`<Backdrop /> calls the onKeyPress function on pressing the ESC key 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  background-color: rgba(0,0,0,0.75);
  z-index: 4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  height: 36px;
  width: 100%;
  min-width: 100px;
  font-weight: 600;
  text-align: center;
  border-radius: 2px;
  color: rgba(255,255,255,1);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,1);
  cursor: pointer;
  font-size: 14px;
  min-width: 28px;
  max-width: 28px;
  height: 28px;
  width: 28px;
  border: none;
}

.c3:disabled {
  color: rgba(255,255,255,0.5);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,0.5);
  cursor: not-allowed;
}

.c3:focus {
  outline: none;
}

.c3 > * {
  pointer-events: none;
}

.c1 {
  background-color: white;
  width: 100%;
  z-index: 8;
  max-height: calc(100% - 60px);
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  position: fixed;
  bottom: 0;
  left: 0;
  padding-right: 8px;
  padding-left: 8px;
}

.bottom-sheet-enter .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.bottom-sheet-leave .c1,
.bottom-sheet-exit .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.bottom-sheet-appear .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.bottom-sheet-enter.bottom-sheet-enter-active .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.bottom-sheet-leave.bottom-sheet-leave-active .c1,
.bottom-sheet-exit.bottom-sheet-exit-active .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.bottom-sheet-appear.bottom-sheet-appear-active .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  margin-left: -8px;
  margin-right: -8px;
  position: relative;
  padding: 16px 24px 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .row--gutterless {
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  margin-left: 0;
  margin-right: 0;
}

.c2.row__button--cancel {
  padding-right: calc(24px + 28px);
  min-height: calc(16px + 28px);
}

.c4 {
  position: absolute;
  top: 16px;
  right: 16px;
  padding: 0;
}

@media screen and (min-width:768px) {
  .c2 {
    margin-left: -12px;
    margin-right: -12px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    padding-left: 32px;
  }

  .c2.row__button--cancel {
    padding-right: calc(32px + 28px);
  }
}

<div
  class="c0 container--overlay"
>
  <div
    aria-modal="true"
    class="c1 content__bottom-sheet"
    role="dialog"
  >
    <div
      class="sc-fzqNqU c2 row__header row__button--cancel"
    >
      <button
        aria-label="Close BottomSheet"
        class="sc-fzoNJl c3 c4 button--cancel"
        role="button"
        size="28"
      >
        <svg
          fill="rgba(38, 38, 38, 1)"
          height="14"
          viewBox="0 0 12 12"
          width="14"
        >
          <path
            d="M6.563 6.203l4.523 4.516a.384.384 0 0 1 .117.281c0 .11-.039.203-.117.281a.378.378 0 0 1-.137.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.136-.09L6 6.766 1.484 11.28a.378.378 0 0 1-.136.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.137-.09A.384.384 0 0 1 .797 11c0-.11.039-.203.117-.281l4.524-4.516L.913 1.68a.384.384 0 0 1-.117-.282c0-.109.039-.203.117-.28A.388.388 0 0 1 1.2 1c.112 0 .207.04.285.117L6 5.633l4.516-4.516A.388.388 0 0 1 10.8 1c.112 0 .207.04.285.117a.384.384 0 0 1 .117.281c0 .11-.039.204-.117.282L6.562 6.203z"
          />
        </svg>
      </button>
    </div>
    <div>
      Europe
    </div>
    <div>
      Africa
    </div>
    <div>
      Asias
    </div>
  </div>
</div>
`;

exports[`<Backdrop /> do not calls the onKeyPress function on pressing any other key than ESC 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  background-color: rgba(0,0,0,0.75);
  z-index: 4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  height: 36px;
  width: 100%;
  min-width: 100px;
  font-weight: 600;
  text-align: center;
  border-radius: 2px;
  color: rgba(255,255,255,1);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,1);
  cursor: pointer;
  font-size: 14px;
  min-width: 28px;
  max-width: 28px;
  height: 28px;
  width: 28px;
  border: none;
}

.c3:disabled {
  color: rgba(255,255,255,0.5);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,0.5);
  cursor: not-allowed;
}

.c3:focus {
  outline: none;
}

.c3 > * {
  pointer-events: none;
}

.c1 {
  background-color: white;
  width: 100%;
  z-index: 8;
  max-height: calc(100% - 60px);
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  position: fixed;
  bottom: 0;
  left: 0;
  padding-right: 8px;
  padding-left: 8px;
}

.bottom-sheet-enter .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.bottom-sheet-leave .c1,
.bottom-sheet-exit .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.bottom-sheet-appear .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.bottom-sheet-enter.bottom-sheet-enter-active .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.bottom-sheet-leave.bottom-sheet-leave-active .c1,
.bottom-sheet-exit.bottom-sheet-exit-active .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.bottom-sheet-appear.bottom-sheet-appear-active .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  margin-left: -8px;
  margin-right: -8px;
  position: relative;
  padding: 16px 24px 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .row--gutterless {
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  margin-left: 0;
  margin-right: 0;
}

.c2.row__button--cancel {
  padding-right: calc(24px + 28px);
  min-height: calc(16px + 28px);
}

.c4 {
  position: absolute;
  top: 16px;
  right: 16px;
  padding: 0;
}

@media screen and (min-width:768px) {
  .c2 {
    margin-left: -12px;
    margin-right: -12px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    padding-left: 32px;
  }

  .c2.row__button--cancel {
    padding-right: calc(32px + 28px);
  }
}

<div
  class="c0 container--overlay"
>
  <div
    aria-modal="true"
    class="c1 content__bottom-sheet"
    role="dialog"
  >
    <div
      class="sc-fzqNqU c2 row__header row__button--cancel"
    >
      <button
        aria-label="Close BottomSheet"
        class="sc-fzoNJl c3 c4 button--cancel"
        role="button"
        size="28"
      >
        <svg
          fill="rgba(38, 38, 38, 1)"
          height="14"
          viewBox="0 0 12 12"
          width="14"
        >
          <path
            d="M6.563 6.203l4.523 4.516a.384.384 0 0 1 .117.281c0 .11-.039.203-.117.281a.378.378 0 0 1-.137.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.136-.09L6 6.766 1.484 11.28a.378.378 0 0 1-.136.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.137-.09A.384.384 0 0 1 .797 11c0-.11.039-.203.117-.281l4.524-4.516L.913 1.68a.384.384 0 0 1-.117-.282c0-.109.039-.203.117-.28A.388.388 0 0 1 1.2 1c.112 0 .207.04.285.117L6 5.633l4.516-4.516A.388.388 0 0 1 10.8 1c.112 0 .207.04.285.117a.384.384 0 0 1 .117.281c0 .11-.039.204-.117.282L6.562 6.203z"
          />
        </svg>
      </button>
    </div>
    <div>
      Europe
    </div>
    <div>
      Africa
    </div>
    <div>
      Asias
    </div>
  </div>
</div>
`;

exports[`<Backdrop /> renders Backdrop correctly 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  background-color: rgba(0,0,0,0.75);
  z-index: 4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  height: 36px;
  width: 100%;
  min-width: 100px;
  font-weight: 600;
  text-align: center;
  border-radius: 2px;
  color: rgba(255,255,255,1);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,1);
  cursor: pointer;
  font-size: 14px;
  min-width: 28px;
  max-width: 28px;
  height: 28px;
  width: 28px;
  border: none;
}

.c3:disabled {
  color: rgba(255,255,255,0.5);
  background-color: transparent;
  border: 1px solid rgba(255,255,255,0.5);
  cursor: not-allowed;
}

.c3:focus {
  outline: none;
}

.c3 > * {
  pointer-events: none;
}

.c1 {
  background-color: white;
  width: 100%;
  z-index: 8;
  max-height: calc(100% - 60px);
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  position: fixed;
  bottom: 0;
  left: 0;
  padding-right: 8px;
  padding-left: 8px;
}

.bottom-sheet-enter .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.bottom-sheet-leave .c1,
.bottom-sheet-exit .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.bottom-sheet-appear .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.bottom-sheet-enter.bottom-sheet-enter-active .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.bottom-sheet-leave.bottom-sheet-leave-active .c1,
.bottom-sheet-exit.bottom-sheet-exit-active .c1 {
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.bottom-sheet-appear.bottom-sheet-appear-active .c1 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms ease-in-out;
  -webkit-transition: transform 300ms ease-in-out;
  transition: transform 300ms ease-in-out;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  box-sizing: border-box;
  margin-left: -8px;
  margin-right: -8px;
  position: relative;
  padding: 16px 24px 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .row--gutterless {
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  margin-left: 0;
  margin-right: 0;
}

.c2.row__button--cancel {
  padding-right: calc(24px + 28px);
  min-height: calc(16px + 28px);
}

.c4 {
  position: absolute;
  top: 16px;
  right: 16px;
  padding: 0;
}

@media screen and (min-width:768px) {
  .c2 {
    margin-left: -12px;
    margin-right: -12px;
  }
}

@media screen and (min-width:768px) {
  .c2 {
    padding-left: 32px;
  }

  .c2.row__button--cancel {
    padding-right: calc(32px + 28px);
  }
}

<div
  class="c0 container--overlay"
>
  <div
    aria-modal="true"
    class="c1 content__bottom-sheet"
    role="dialog"
  >
    <div
      class="sc-fzqNqU c2 row__header row__button--cancel"
    >
      <button
        aria-label="Close BottomSheet"
        class="sc-fzoNJl c3 c4 button--cancel"
        role="button"
        size="28"
      >
        <svg
          fill="rgba(38, 38, 38, 1)"
          height="14"
          viewBox="0 0 12 12"
          width="14"
        >
          <path
            d="M6.563 6.203l4.523 4.516a.384.384 0 0 1 .117.281c0 .11-.039.203-.117.281a.378.378 0 0 1-.137.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.136-.09L6 6.766 1.484 11.28a.378.378 0 0 1-.136.09.417.417 0 0 1-.297 0 .378.378 0 0 1-.137-.09A.384.384 0 0 1 .797 11c0-.11.039-.203.117-.281l4.524-4.516L.913 1.68a.384.384 0 0 1-.117-.282c0-.109.039-.203.117-.28A.388.388 0 0 1 1.2 1c.112 0 .207.04.285.117L6 5.633l4.516-4.516A.388.388 0 0 1 10.8 1c.112 0 .207.04.285.117a.384.384 0 0 1 .117.281c0 .11-.039.204-.117.282L6.562 6.203z"
          />
        </svg>
      </button>
    </div>
    <div>
      Europe
    </div>
    <div>
      Africa
    </div>
    <div>
      Asias
    </div>
  </div>
</div>
`;
